### ISSUE TITLE
gskit: compile failures in x509asn1.c

### ISSUE THREAD

**Issue by jonrumsey** [curl/curl#10237]
### I did this

Compiled libCurl 7.87.0 with GSKit support enabled after pulling in https://github.com/curl/curl/pull/10207 to address https://github.com/curl/curl/issues/10163. Some compilation warnings and errors still occur;

```
x509asn1.c
..\lib\vtls\x509asn1.c(1283): warning C4013: 'Curl_ssl_cf_get_primary_config' undefined; assuming extern returning int
..\lib\vtls\x509asn1.c(1312): error C2037: left of 'hostname' specifies undefined struct/union 'ssl_connect_data'
..\lib\vtls\x509asn1.c(1312): error C2168: 'strlen': too few actual parameters for intrinsic function
..\lib\vtls\x509asn1.c(1316): error C2065: 'conn': undeclared identifier
..\lib\vtls\x509asn1.c(1316): error C2223: left of '->bits' must point to struct/union
..\lib\vtls\x509asn1.c(1316): error C2037: left of 'hostname' specifies undefined struct/union 'ssl_connect_data'
..\lib\vtls\x509asn1.c(1316): warning C4133: 'function': incompatible types - from 'in6_addr *' to 'const char *'
..\lib\vtls\x509asn1.c(1316): error C2198: 'Curl_inet_pton': too few arguments for call
..\lib\vtls\x509asn1.c(1320): error C2037: left of 'hostname' specifies undefined struct/union 'ssl_connect_data'
..\lib\vtls\x509asn1.c(1320): warning C4133: 'function': incompatible types - from 'in6_addr *' to 'const char *'
..\lib\vtls\x509asn1.c(1320): error C2198: 'Curl_inet_pton': too few arguments for call
..\lib\vtls\x509asn1.c(1355): error C2037: left of 'hostname' specifies undefined struct/union 'ssl_connect_data'
..\lib\vtls\x509asn1.c(1355): warning C4047: 'function': 'const char *' differs in levels of indirection from 'size_t'
..\lib\vtls\x509asn1.c(1355): warning C4024: 'Curl_cert_hostcheck': different types for formal and actual parameter 3
..\lib\vtls\x509asn1.c(1354): error C2198: 'Curl_cert_hostcheck': too few arguments for call
..\lib\vtls\x509asn1.c(1362): warning C4389: '==': signed/unsigned mismatch
..\lib\vtls\x509asn1.c(1373): error C2037: left of 'dispname' specifies undefined struct/union 'ssl_connect_data'
..\lib\vtls\x509asn1.c(1378): error C2037: left of 'dispname' specifies undefined struct/union 'ssl_connect_data'
..\lib\vtls\x509asn1.c(1415): error C2037: left of 'hostname' specifies undefined struct/union 'ssl_connect_data'
..\lib\vtls\x509asn1.c(1415): warning C4047: 'function': 'const char *' differs in levels of indirection from 'size_t'
..\lib\vtls\x509asn1.c(1415): warning C4024: 'Curl_cert_hostcheck': different types for formal and actual parameter 3
..\lib\vtls\x509asn1.c(1414): error C2198: 'Curl_cert_hostcheck': too few arguments for call
..\lib\vtls\x509asn1.c(1422): error C2037: left of 'dispname' specifies undefined struct/union 'ssl_connect_data'
..\lib\vtls\x509asn1.c(1283): warning C4047: 'initializing': 'ssl_primary_config *' differs in levels of indirection from 'int'
```

Adding an include of "vtls/vtls.h" removes most of these warnings;

```
x509asn1.c
..\lib\vtls\x509asn1.c(1317): error C2065: 'conn': undeclared identifier
..\lib\vtls\x509asn1.c(1317): error C2223: left of '->bits' must point to struct/union
..\lib\vtls\x509asn1.c(1363): warning C4389: '==': signed/unsigned mismatch
```

Changing "conn" to "cf->conn" and casting (name.end - name.beg) to a size_t removes the last of these errors and warnings.

### I expected the following

Clean build.

### curl/libcurl version

7.87.0

### operating system

Various platforms using GSKit (including OS400).

**Comment by bagder** [comment]
Related: #10201

**Comment by bagder** [comment]
Fixed via #10238
